src/common/KokkosKernels_Sorting.hpp

```diff
@@ -43,15 +43,664 @@
 */
 #ifndef _KOKKOSKERNELS_SORTING_HPP
 #define _KOKKOSKERNELS_SORTING_HPP
+
 #include "Kokkos_Core.hpp"
-#include "Kokkos_Atomic.hpp"
-#include "Kokkos_ArithTraits.hpp"
-#include "impl/Kokkos_Timer.hpp"
+#include "KokkosKernels_SimpleUtils.hpp"  //for kk_exclusive_parallel_prefix_sum
+#include "KokkosKernels_ExecSpaceUtils.hpp"//for kk_is_gpu_exec_space
 #include <type_traits>
 
 namespace KokkosKernels {
+
+namespace Impl {
+  template<typename Value>
+  struct DefaultComparator
+  {
+    KOKKOS_INLINE_FUNCTION bool operator()(const Value lhs, const Value rhs) const
+    {
+      return lhs < rhs;
+    }
+  };
+}
+
+// ----------------------------------
+// CRS matrix/graph sorting utilities
+// ----------------------------------
+
+// The sort_crs* functions sort the adjacent column list for each row into ascending order.
+
+template<typename execution_space, typename rowmap_t, typename entries_t, typename values_t>
+void sort_crs_matrix(const rowmap_t& rowmap, const entries_t& entries, const values_t& values);
+
+template <typename crsMat_t>
+void sort_crs_matrix(const crsMat_t& A);
+
+template<typename execution_space, typename rowmap_t, typename entries_t>
+void sort_crs_graph(const rowmap_t& rowmap, const entries_t& entries);
```

Comment:
I would argue that we could have one more function signature that takes a StaticCrsGraph as input and only has one template parameter but that's probably not the most critical.